language/move-ir-compiler/ir-to-bytecode/syntax/src/syntax.rs

```diff
@@ -916,29 +916,72 @@ fn parse_assign_(tokens: &mut Lexer) -> Result<Cmd_, ParseError<Loc, anyhow::Err
     }
     consume_token(tokens, Tok::Equal)?;
     let e = parse_exp(tokens)?;
-    Ok(Cmd_::Assign(lvalues, e))
+    Ok(Statement_::Assign(lvalues, e))
 }
 
 fn parse_unpack_(
     tokens: &mut Lexer,
     name: Symbol,
     type_actuals: Vec<Type>,
-) -> Result<Cmd_, ParseError<Loc, anyhow::Error>> {
+) -> Result<Statement_, ParseError<Loc, anyhow::Error>> {
     consume_token(tokens, Tok::LBrace)?;
     let bindings = parse_comma_list(tokens, &[Tok::RBrace], parse_field_bindings, true)?;
     consume_token(tokens, Tok::RBrace)?;
     consume_token(tokens, Tok::Equal)?;
     let e = parse_exp(tokens)?;
-    Ok(Cmd_::Unpack(
+    Ok(Statement_::Unpack(
         StructName(name),
         type_actuals,
         bindings.into_iter().collect(),
         Box::new(e),
     ))
 }
 
-fn parse_cmd_(tokens: &mut Lexer) -> Result<Cmd_, ParseError<Loc, anyhow::Error>> {
+/// Parses a statement.
+fn parse_statement_(tokens: &mut Lexer) -> Result<Statement_, ParseError<Loc, anyhow::Error>> {
     match tokens.peek() {
+        Tok::Abort => {
+            tokens.advance()?;
+            let val = if tokens.peek() == Tok::Semicolon {
+                None
```

Comment:
It was optional before. As above, I didn't change this as part of this pull request. I think it shouldn't be optional, and AFAICT [this](https://github.com/diem/diem/blob/21709d41776bbfef6bda89662a06d2a4dc42a840/language/bytecode-verifier/transactional-tests/tests/stack_usage_verifier/abort_negative_stack_size.mvir) is the only test that uses it without an expression. Still, I think this could be changed in an upcoming pull request; no need to change more than necessary in this one since it's so large and it's been in review for so long.

Reply:
Ah I see now. I sort of forgot when reviewing this that the argument to abort was dynamic. We could maybe do something like `abort ();` instead in the future to clean up the parser